tests/unit/test_local_engine.py

```diff
@@ -0,0 +1,215 @@
+
+
+# All tests in this file require polars to be installed to pass.
+@pytest.fixture(scope="module")
+def polars_session():
+    from . import polars_session
+
+    return polars_session.TestSession()
+
+
+@pytest.fixture(scope="module")
+def test_frame() -> pd.DataFrame:
+    df = pd.DataFrame(
+        {
+            "int1": pd.Series([1, 2, 3], dtype="Int64"),
+            "int2": pd.Series([-10, 20, 30], dtype="Int64"),
+            "bools": pd.Series([True, None, False], dtype="boolean"),
+            "strings": pd.Series(["b", "aa", "ccc"], dtype="string[pyarrow]"),
+            "intLists": pd.Series(
+                [[1, 2, 3], [4, 5, 6, 7], None],
+                dtype=pd.ArrowDtype(pa.list_(pa.int64())),
+            ),
+        },
+    )
+    df.index = df.index.astype("Int64")
+    return df
+
+
+# These tests should be unit tests, but Session object is tightly coupled to BigQuery client.
+@skip_legacy_pandas
+def test_polars_local_engine_add(
+    test_frame: pd.DataFrame, polars_session: bigframes.Session
+):
+    pd_df = test_frame
+    bf_df = bpd.DataFrame(pd_df, session=polars_session)
+
+    bf_result = (bf_df["int1"] + bf_df["int2"]).to_pandas()
+    pd_result = pd_df.int1 + pd_df.int2
+    pandas.testing.assert_series_equal(bf_result, pd_result)
+
+
+@skip_legacy_pandas
+def test_polars_local_engine_order_by(test_frame: pd.DataFrame, polars_session):
+    pd_df = test_frame
+    bf_df = bpd.DataFrame(pd_df, session=polars_session)
+
+    bf_result = bf_df.sort_values("strings").to_pandas()
+    pd_result = pd_df.sort_values("strings")
+    pandas.testing.assert_frame_equal(bf_result, pd_result)
+
+
+@skip_legacy_pandas
+def test_polars_local_engine_filter(test_frame: pd.DataFrame, polars_session):
+    pd_df = test_frame
+    bf_df = bpd.DataFrame(pd_df, session=polars_session)
+
+    bf_result = bf_df.filter(bf_df["int2"] >= 1).to_pandas()
+    pd_result = pd_df.filter(pd_df["int2"] >= 1)  # type: ignore
+    pandas.testing.assert_frame_equal(bf_result, pd_result)
+
+
+@skip_legacy_pandas
+def test_polars_local_engine_reset_index(test_frame: pd.DataFrame, polars_session):
+    pd_df = test_frame
+    bf_df = bpd.DataFrame(pd_df, session=polars_session)
+
+    bf_result = bf_df.reset_index().to_pandas()
+    pd_result = pd_df.reset_index()
+    # pd default index is int64, bf is Int64
+    pandas.testing.assert_frame_equal(bf_result, pd_result, check_index_type=False)
+
+
+@skip_legacy_pandas
+def test_polars_local_engine_join_binop(polars_session):
+    pd_df_1 = pd.DataFrame({"colA": [1, None, 3], "colB": [3, 1, 2]}, index=[1, 2, 3])
+    pd_df_2 = pd.DataFrame(
+        {"colA": [100, 200, 300], "colB": [30, 10, 40]}, index=[2, 1, 4]
+    )
+    bf_df_1 = bpd.DataFrame(pd_df_1, session=polars_session)
+    bf_df_2 = bpd.DataFrame(pd_df_2, session=polars_session)
+
+    bf_result = (bf_df_1 + bf_df_2).to_pandas()
+    pd_result = pd_df_1 + pd_df_2
+    # Sort by index because ordering logic isn't quite consistent yet
```

Comment:
Maybe fixed now?

Reply:
eh, I made it consitent with sql engine now, but still not with pandas. Pandas order I think is a bit of a runtime decision